### PR TITLE
Refactoring `getPolicyPlacementDecisions`

### PR DESCRIPTION
**Pull request by JustinKuli**
The first commit is a collection of small-ish things I noticed while looking for possible inefficiencies in the propagator. The second has the "meat" of this PR - read its commit message for more details.

While this PR might slightly improve the performance of the propagator, it's unlikely to be a major step forward.

### CODE REVIEW

File: controllers/common/common.go [hunk 1]
```diff
@@ -60,24 +62,25 @@ func IsReplicatedPolicy(c client.Client, policy client.Object) (bool, error) {
...
+func IsForPolicyOrPolicySet(pb *policiesv1.PlacementBinding) bool {
+	if pb == nil {
+		return false
+	}
 
-			break
+	for _, subject := range pb.Subjects {
+		if subject.APIGroup == policiesv1.SchemeGroupVersion.Group &&
+			(subject.Kind == policiesv1.Kind || subject.Kind == policiesv1.PolicySetKind) {
+			return true
```
Comment (yiraeChristineKim) [inline controllers/common/common.go:75]: I prefer adding `break`, if we find policy... why did you remove this

File: controllers/common/common.go [hunk 2]
```diff
@@ -60,24 +62,25 @@ func IsReplicatedPolicy(c client.Client, policy client.Object) (bool, error) {
...
 	}
 
-	return found
+	return false
 }
 
-// IsPbForPoicySet compares group and kind with policyset group and kind for given pb
-func IsPbForPoicySet(pb *policiesv1.PlacementBinding) bool {
+// IsPbForPolicySet compares group and kind with policyset group and kind for given pb
+func IsPbForPolicySet(pb *policiesv1.PlacementBinding) bool {
```
Comment (yiraeChristineKim) [inline controllers/common/common.go:83]: good catch but Poicy set was so cute! Thanks

File: controllers/common/common.go (hunk 1 above)
Comment (JustinKuli) [reply controllers/common/common.go:75]: Why is `break` better? I feel like it's the same

File: controllers/common/common.go [hunk 3]
```diff
@@ -92,6 +95,48 @@ func IsPbForPoicySet(pb *policiesv1.PlacementBinding) bool {
 	return found
 }
 
+// PoliciesInPlacementBinding returns a list of the Policies that are either direct subjects of
+// the given PlacementBinding, or are in PolicySets that are subjects of the PlacementBinding.
+// The list items are not guaranteed to be unique (for example if a policy is in multiple sets).
+func PoliciesInPlacementBinding(
```
Comment (yiraeChristineKim) [inline controllers/common/common.go:101]: Get + policiesInPlacemantBiding?

File: controllers/propagator/propagation.go [hunk 4]
```diff
@@ -1229,30 +1096,24 @@ func isConfigurationPolicy(policyT *policiesv1.PolicyTemplate) bool {
...
-func (r *PolicyReconciler) isPolicySetSubject(instance *policiesv1.Policy, subject policiesv1.Subject) bool {
-	log := log.WithValues("policyName", instance.GetName(), "policyNamespace", instance.GetNamespace())
-
-	if subject.APIGroup == policiesv1.SchemeGroupVersion.Group &&
-		subject.Kind == policiesv1.PolicySetKind {
-		policySetNamespacedName := types.NamespacedName{
-			Name:      subject.Name,
-			Namespace: instance.GetNamespace(),
-		}
+func (r *PolicyReconciler) PolicySetHasPolicy(policySetName, policyName, namespace string) bool {
```
Comment (yiraeChristineKim) [inline controllers/propagator/propagation.go:1099]: function name? maybe hasPolicySetInPolicSet hm... weird verb is not first but not a big deal

File: controllers/propagator/propagation.go [hunk 5]
```diff
@@ -1229,30 +1096,24 @@ func isConfigurationPolicy(policyT *policiesv1.PolicyTemplate) bool {
...
+		return false
+	}
 
-		for _, plc := range policySet.Spec.Policies {
-			if string(plc) == instance.GetName() {
-				return true
-			}
+	for _, plc := range policySet.Spec.Policies {
+		if string(plc) == policyName {
+			return true
```
Comment (yiraeChristineKim) [inline controllers/propagator/propagation.go:1116]: add `break` for performance?

File: controllers/common/common.go (hunk 1 above)
Comment (yiraeChristineKim) [reply controllers/common/common.go:75]: oh yeah same!

Comment (yiraeChristineKim) [reply controllers/common/common.go:75]: I was working with javascript so confused lol

File: controllers/propagator/propagation.go (hunk 4 above)
Comment (JustinKuli) [reply controllers/propagator/propagation.go:1099]: Maybe `IsPolicyInPolicySet`?

Comment (yiraeChristineKim) [reply controllers/propagator/propagation.go:1099]: Good!

**Comment by JustinKuli** [comment]
Turns out (of course) some of the stuff I thought I could just remove did have a subtle effect... the new updates look pretty different from what I had originally, but with helpful comments when there is something tricky going on.

File: controllers/common/common.go (hunk 3 above)
Comment (JustinKuli) [reply controllers/common/common.go]: Renamed to `GetPoliciesInPlacementBinding`

**Review by yiraeChristineKim — APPROVED** [review]
